internal/ethapi/transaction_args.go

```diff
@@ -84,7 +84,7 @@ func (args *TransactionArgs) setDefaults(ctx context.Context, b Backend) error {
 	// need to consult the chain for defaults. It's definitely a London tx.
 	if args.MaxPriorityFeePerGas == nil || args.MaxFeePerGas == nil {
 		// In this clause, user left some fields unspecified.
-		if b.ChainConfig().IsLondon(head.Number) && args.GasPrice == nil {
+		if b.ChainConfig().IsBanach(head.Number) && args.GasPrice == nil {
```

Comment:
Behavior after London and before Banach:

- Original behavior: Enter `if` branch, set `MaxPriorityFeePerGas` and `MaxFeePerGas`.
- New Behavior: Enter `else` branch, fail if user set any 1559 fields.

Slightly changed behaviour, but I think it's acceptable. If the user used `eth_call` with EIP-1559 fields before, their transaction would have failed anyway.